packages/itwinui-react/src/core/ComboBox/ComboBoxMenu.tsx

```diff
@@ -61,7 +61,10 @@ const VirtualizedComboBoxMenu = React.forwardRef(
 
     const surfaceStyles = {
       minWidth,
-      maxWidth: `min(${minWidth * 2}px, 90vw)`,
+
+      // set as constant because we don't want it shifting when items are unmounted
+      maxWidth: minWidth,
```

Comment:
Have you checked how it looks when virtualization is enabled and the item which is not rendered initially is much wider than the calculated width?

Reply:
![](https://user-images.githubusercontent.com/9084735/211355871-eb171874-7de4-49bd-a9a6-6fbfa9123c4d.png)

Reply:
oh ok.. so it just never recalculates and hardcodes the width as it is calculated from the initial visible list.